### ISSUE TITLE
ECR Login fails in Packer 1.8.3

### ISSUE THREAD

**Issue by evanstachowiak** [hashicorp/packer-plugin-docker#110]
#### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

When filing a bug, please include the following headings if possible. Any
example text in this template can be deleted.

#### Overview of the Issue

Without any changes to the Packer HCL template, Packer builds are now failing with the error:
```
Error fetching ECR credentials: failed to parse the ECR URL: account-id.dkr.ecr.us-east-1.amazonaws.com
==> build.docker.shift: parse "account-id.dkr.ecr.us-east-1.amazonaws.com": invalid URI for request
==> build.docker.shift: it should be either of the form `public.ecr.aws/<registry_alias>/<registry_name>` or `<account number>.dkr.ecr.<region>.amazonaws.com`
Build 'build.docker.shift' errored after 76 milliseconds 463 microseconds: Error fetching ECR credentials: failed to parse the ECR URL: account-id.dkr.ecr.us-east-1.amazonaws.com
parse "account-id.dkr.ecr.us-east-1.amazonaws.com": invalid URI for request
it should be either of the form `public.ecr.aws/<registry_alias>/<registry_name>` or `<account number>.dkr.ecr.<region>.amazonaws.com`
```
(I have substituted the account id number with "account-id" string above)
#### Reproduction Steps

Steps to reproduce this issue

`packer build` with an ECR login.

### Packer version
1.8.3
From `packer version`

### Operating system and Environment details
Ubuntu 22.04, macOS 12.4

**Comment by evanstachowiak** [comment]
Hi @nywilken , thanks for the quick reply. if i add `https://` I get the error: `invalid reference format`

**Comment by leoxlin** [comment]
We were able to reproduce this in packer version `1.8.0`, `1.8.1` and `1.8.2` alongs side packer-plugin-docker version `1.0.7` using the ECR url `redacted-account-id.dkr.ecr.us-east-1.amazonaws.com`

Once we downgraded to `1.0.6` the issue is gone so it seems like this regression is introduced in the `1.0.7` version.

**Comment by devashish-patel** [comment]
@leoxlin @evanstachowiak - I have added the fix in PR #111, I will release the fix in the upcoming version in a few hours. Thanks for your patience.

**Comment by devashish-patel** [comment]
We have released the fix for this problem in `v1.0.8`.. please update the plugin and everything should start working as normal. Please let us know if you find something abnormal.

Thanks a lot for reporting the issue!
